### ISSUE TITLE
Calling the "error" function causes a segmentation fault on the CentOS implmentation of Julia 6.2

### ISSUE THREAD

**Issue by ahalwright** [JuliaLang/julia#27190]
The following causes a segmentation fault when run the command line in CentOS 7.3 using julia v 6.2:

julia -E 'error("error")'

This is the simplest example that I have found---I have other examples, especially calling julia with the "-L" option.

Here is my OS information:
CentOS Linux release 7.3.1611 (Core)
Linux version 3.10.0-514.26.2.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-11) (GCC) ) #1 SMP Tue Jul 4 15:04:05 UTC 2017

Here is the full Julia error message:

$ julia -E 'error("error")'

signal (11): Segmentation fault
while loading no file, in expression starting on line 0
StringRef at /buildworker/worker/package_linux64/build/usr/include/llvm/ADT/StringRef.h:69 [inlined]
getDebuglink at /buildworker/worker/package_linux64/build/src/debuginfo.cpp:904 [inlined]
jl_dylib_DI_for_fptr at /buildworker/worker/package_linux64/build/src/debuginfo.cpp:1253
jl_getDylibFunctionInfo at /buildworker/worker/package_linux64/build/src/debuginfo.cpp:1397 [inlined]
jl_getFunctionInfo at /buildworker/worker/package_linux64/build/src/debuginfo.cpp:1633
jl_lookup_code_address at /buildworker/worker/package_linux64/build/src/stackwalk.c:348
lookup at ./stacktraces.jl:125
collect_to! at ./array.jl:508
_collect at ./array.jl:489
stacktrace at ./stacktraces.jl:151
display_error at ./client.jl:126
unknown function (ip: 0x7f3a89d5063d)
jl_call_fptr_internal at /buildworker/worker/package_linux64/build/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /buildworker/worker/package_linux64/build/src/julia_internal.h:358 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:1926
display_error at ./client.jl:140
unknown function (ip: 0x7f3a89d500a6)
jl_call_fptr_internal at /buildworker/worker/package_linux64/build/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /buildworker/worker/package_linux64/build/src/julia_internal.h:358 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:1926
do_call at /buildworker/worker/package_linux64/build/src/interpreter.c:75
eval at /buildworker/worker/package_linux64/build/src/interpreter.c:242
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:539
jl_toplevel_eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:511
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:571
jl_toplevel_eval_in at /buildworker/worker/package_linux64/build/src/builtins.c:496
eval at ./boot.jl:235
unknown function (ip: 0x7f3a9e594d2f)
jl_call_fptr_internal at /buildworker/worker/package_linux64/build/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /buildworker/worker/package_linux64/build/src/julia_internal.h:358 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:1926
_start at ./client.jl:417
unknown function (ip: 0x7f3a9e5e6d28)
jl_call_fptr_internal at /buildworker/worker/package_linux64/build/src/julia_internal.h:339 [inlined]
jl_call_method_internal at /buildworker/worker/package_linux64/build/src/julia_internal.h:358 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:1926
Segmentation fault (core dumped)

**Comment by nalimilan** [comment]
Thanks for the report. How did you install Julia?

**Comment by ahalwright** [comment]
I downloaded  julia-0.6.2-linux-x86_64.tar.gz  from  https://julialang.org/downloads/  and am running the corresponding binaries.  Would you like me to install differently?

Alden Wright

**Comment by ararslan** [comment]
Interesting given that, if I remember correctly, we build our Linux binaries on CentOS...

**Comment by ahalwright** [comment]
I enabled EPEL and installed version 0.7.0-DEV.  The example that I posted does not give a segmentation fault in this version.  I will try other examples over the next few days.

**Comment by nalimilan** [comment]
> I downloaded julia-0.6.2-linux-x86_64.tar.gz from https://julialang.org/downloads/ and am running the corresponding binaries. Would you like me to install differently?

No, that's the official method.



> I enabled EPEL and installed version 0.7.0-DEV. The example that I posted does not give a segmentation fault in this version. I will try other examples over the next few days.

You mean using the Copr package? Can you also try with the one at https://julialangnightlies-s3.julialang.org/bin/linux/x64/julia-latest-linux64.tar.gz?

**Comment by ahalwright** [comment]
I am getting segmentation fault errors with the EPEL version (installed by "yum install"), but not the "julianightlies" version of the previous comment.  Both versions are "0.7.0-DEV".  I have not yet isolated the specific function that causes the segmentation fault in the EPEL version, but I will work on this.

**Comment by ahalwright** [comment]
I found that whether there was a segmentation fault depended on where I was running Julia rather than on which executable I was running.  I discovered that that when there was an empty file named "julia" in the folder where I was running julia, I would get the segmentation fault.  How this empty file was being created, I don't know.  In any case, the issue has something to do with the environment where the executable is being run rather than a direct flaw in the executable.
